interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/FragmentComponentDefinitionExtractor.scala

```diff
@@ -101,30 +104,52 @@ class FragmentComponentDefinitionExtractor(
 
   private def toParameter(
       componentConfig: SingleComponentConfig
-  )(p: FragmentParameter): Writer[List[FragmentParamClassLoadErrorData], Parameter] = {
-    val paramName = p.name
-    p.typ
+  )(
+      fragmentParameter: FragmentParameter
+  )(implicit nodeId: NodeId): Writer[List[FragmentParamClassLoadErrorData], Parameter] = {
+    fragmentParameter.typ
       .toRuntimeClass(classLoader)
       .map(Typed(_))
       .map(Writer.value[List[FragmentParamClassLoadErrorData], TypingResult])
       .getOrElse(
         Writer
           .value[List[FragmentParamClassLoadErrorData], TypingResult](Unknown)
-          .tell(List(FragmentParamClassLoadErrorData(paramName, p.typ.refClazzName)))
+          .tell(List(FragmentParamClassLoadErrorData(fragmentParameter.name, fragmentParameter.typ.refClazzName)))
       )
-      .map(toParameter(componentConfig, paramName, _))
+      .map(toParameter(componentConfig, _, fragmentParameter))
   }
 
-  private def toParameter(componentConfig: SingleComponentConfig, paramName: String, typ: typing.TypingResult) = {
-    val config          = componentConfig.params.flatMap(_.get(paramName)).getOrElse(ParameterConfig.empty)
+  private def toParameter(
+      componentConfig: SingleComponentConfig,
+      typ: typing.TypingResult,
+      fragmentParameter: FragmentParameter
+  )(implicit nodeId: NodeId): Parameter = {
+    val config          = componentConfig.params.flatMap(_.get(fragmentParameter.name)).getOrElse(ParameterConfig.empty)
     val parameterData   = ParameterData(typ, Nil)
     val extractedEditor = EditorExtractor.extract(parameterData, config)
+
+    val expressionConfig   = ProcessDefinitionBuilder.empty.expressionConfig
+    val expressionCompiler = ExpressionCompiler.withOptimization(classLoader, null, expressionConfig, null)
```

Comment:
we can use simple dict registry implementation, but `typeDefinitionSet` mus be real, from model. otherwise e.g. method calling won't work

Reply:
It wasn't easy, but it's done :)